File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.cpp
#include <string>
#include <vector>

using namespace std;

int isPrime(int n) {
    if(n < 2) return 0;
    if(n <= 3) return 1;
    
    if((n % 2 == 0) || (n % 3 == 0)) return false;
    
    for(int i=5; (i*i )<=n; i+=6)
        if((n % i == 0) || (n % (i+2) == 0)) return 0;
    
    return 1;
}

int solution(int n) {
    int answer = 0;
    for(int i=1; i<=n; ++i) {
        if(isPrime(i)) ++answer;
    }
    return answer;
}

File: 프로그래머스/lv1/12921. 소수 찾기/소수 찾기.rb
def prime?(n)
    if n < 2 then return false end
    if n <= 3 then return true end
    
    if (n % 2 == 0) or (n % 3 == 0) then return false end
    
    i = 5
    while (i*i) <= n
        if (n % i == 0) or (n % (i+2) == 0) then return false end
        i += 6
    end
    
    true
end
        
def solution(n)
    cnt = 0
    for i in (1..n)
        if prime? i then cnt +=1 end
    end
    cnt
end